Home work UI

TimurBikaev opened the pull request:
Жаль времени маловато, тут непаханное поле для тестиков))

Написал основные: Позитивные - создание проектов и билдконфигов обоими способами(Url репозитория и вручную)
И по несколько негативных (пустой урл репозитория, некорректный урл репозитория, приватный репозиторий без кредов...), а также пустые обязательные поля Name и ID для обеих сущностей.
Надеюсь, что будет достаточно для базового дз.

Стабильность: прогонял по отдельности и пачками несколько раз. Выявил ограничение на 100 билдов, а также ограничение имени в сообщениях успеха. Поправил некоторые флакки ожиданиями.

Спасибо за такой ценный опыт!

tbikaev-realtrac committed:
Модуль UI подготовка к ДЗ

tbikaev-realtrac committed:
Созданы позитивные и негативные тесты для создания Проектов и БилдКонфигов через UI.
Добавлены классы страниц, элементы и методы.

Удален BaseApiTest.java
Рефакторинг

nobugs-me commented on src/main/java/com/example/teamcity/ui/pages/favorites/FavoritesPage.java:
```diff
@@ -0,0 +1,35 @@
+
+
+
+
```
Минор: пустые строки.

nobugs-me commented on src/test/java/com/example/teamcity/ui/Test_CreateNewBuildConfig.java:
```diff
@@ -0,0 +1,226 @@
+        switchToManually();
+
+        //создание через Manually
+        CreateNewBuildConfig.createBuildConfigByManually_name(buldConfigName);
```
Мы не используем нижние подчеркивания в именах методов в Java (да и в других местах тоже) - только CamelCase - `createBuildConfigByManuallyName`

nobugs-me commented on src/test/java/com/example/teamcity/ui/Test_CreateNewProject.java:
```diff
@@ -0,0 +1,138 @@
+        new ProjectsPage().open()
+                .getSubprojects() // забираем значения subprojects
+                .stream().reduce((first, second) -> second).get() //забираем последний созданный элемент
+                .getHeader().shouldHave(Condition.text(testData.getProject().getName()));// проверяем появление имени проекта
```
Отлично! Теперь нужно добавить проверку интеграции - что проект успешно создан на уровне API.

nobugs-me commented on src/test/java/com/example/teamcity/ui/Test_CreateNewProject.java:
```diff
@@ -0,0 +1,138 @@
+
+        //Проверяем сообщение об успехе на след.странице
+        checkSuccessCreateProject(projectName);
+        //у сообщения ограниченная вместимость по символам, поэтому полностью может не уместиться!
```
Здесь также проверка интеграции.

nobugs-me reviewed (approved):
Привет! 

Спасибо за выполненное ДЗ! 

Все реализовано корректно, осталось только добавить важную проверку интеграции - что сущности были созданы на уровне API.

Также обрати внимание на форматтинг - во многих классах пропущены строки или используется некорректное именование через нижнее подчеркивание. 

И отдельно спасибо за комментарии - видна детализированная работа. Если будет интересно поучаствовать в будущих потоках, как часть команды кураторов - обязательно напиши мне в ТГ.

tbikaev-realtrac committed:
Добавил проверки созданных сущностей "проект" и "билдКонфиг" разными способами. Долго не мог найти способ дергать билдКонфиг без его имени, поэтому изобрел другой GET-запрос .get(projectId + "_Build" + buildName + "?mode=builds") --есть подробный коммент. Для этого пришлось добавить другой get и наплодить UncheckedBuildConfig -- позже приведу в порядок.

Позже уже ребята в чате подсказали, как дергать билдКонфиг по локатору имени (нужно было убрать "/id:" и сразу ставить имя в запросе)). Тоже добавил такой способ.

Также провел рефакторинг кода, сменил названия классов со snake_case на camelCase (я один авто в компании, живу без ревью кода).
Устранил пустые строки (хотя, постоянно делаю ctrl+alt+L).

TimurBikaev commented:
Благодарю за приглашение в Команду кураторов на будущих потоках)) С радостью впишусь, чтобы самому продолжить оттачивать полученные здесь скиллы и помогать другим в этом!

Если честно, у самого крутилась мысль каким-то образом напроситься к тебе в любом качестве подмастерьем и перенимать твой опыт и философию, т.к. давно подписан на тебя на ютубе (на рус и англ.языках) и в подкасте "Вроде в проде", недавно еще Кружок и Адвент-календарь посмотрел)) и разделяю всю твою движуху
